JustSaying.AwsTools/MessageHandling/SqsPublisher.cs

```diff
@@ -19,13 +21,23 @@ public SqsPublisher(RegionEndpoint region, string queueName, IAmazonSQS client,
             _serialisationRegister = serialisationRegister;
         }
 
-        public void Publish(Message message)
+        public async Task Publish(Message message)
         {
-            _client.SendMessage(new SendMessageRequest
+            var request = new SendMessageRequest
+                {
+                    MessageBody = GetMessageInContext(message),
+                    QueueUrl = Url
+                };
+
+            try
+            {
+                await _client.SendMessageAsync(request)
+                    .ConfigureAwait(false);
+            }
+            catch (Exception ex)
             {
-                MessageBody = GetMessageInContext(message),
-                QueueUrl = Url
-            });
+                throw new Exception($"Failed to publish message to SQS. QueueUrl: {request.QueueUrl} MessageBody: {request.MessageBody}", ex);
```

Comment:
This is good cause it shields the complexity from the consuming code (JustSaying is an abstraction), and wrapping keeps the information (https://msdn.microsoft.com/en-us/library/ms229049(v=vs.100).aspx)

It would be nice to have our own exception, even as generic as ```PublishException``` and have it in the method xml summary, so others devs will see it as they use it, if they see just ```Exception```, they may be hesitant to catch that as they may catch general exceptions local arround that call. That can come later though.